packages/sdk/identity/src/odis/identifier.ts

```diff
@@ -0,0 +1,217 @@
+export async function getBlindedIdentifier(
+  identifier: string,
+  identifierPrefix: string,
+  blsBlindingClient: BlsBlindingClient,
+  seed?: Buffer
+): Promise<string> {
+  debug('Retrieving blinded message')
+  // phone number identifiers don't have prefixes in the blinding stage
+  // to maintain backwards compatibility wih ASv1
+  const base64Identifier =
+    identifierPrefix === IdentifierPrefix.PHONE_NUMBER
+      ? Buffer.from(identifier).toString('base64')
+      : Buffer.from(getPrefixedIdentifier(identifier, identifierPrefix)).toString('base64')
+  return blsBlindingClient.blindMessage(base64Identifier, seed)
+}
+
+/**
+ * Query ODIS for the blinded signature
+ * Response can be passed into getObfuscatedIdentifierFromSignature
+ * to retrieve the on-chain identifier
+ */
+export async function getBlindedIdentifierSignature(
+  account: string,
+  signer: AuthSigner,
+  context: ServiceContext,
+  base64BlindedMessage: string,
+  clientVersion?: string,
+  sessionID?: string,
+  keyVersion?: number,
+  endpoint?: CombinerEndpointPNP.LEGACY_PNP_SIGN | CombinerEndpointPNP.PNP_SIGN
+): Promise<string> {
+  const body: SignMessageRequest = {
+    account,
+    blindedQueryPhoneNumber: base64BlindedMessage,
+    version: clientVersion,
+    authenticationMethod: signer.authenticationMethod,
+    sessionID,
+  }
+
+  const response = await queryOdis(
+    body,
+    context,
+    endpoint ?? CombinerEndpointPNP.PNP_SIGN,
+    SignMessageResponseSchema,
+    {
+      [KEY_VERSION_HEADER]: keyVersion?.toString(),
+      Authorization: await getOdisPnpRequestAuth(body, signer),
+    }
+  )
+
+  if (!response.success) {
+    throw new Error(response.error)
+  }
+
+  return response.signature
+}
+
+/**
+ * Unblind the response and return the on-chain identifier
+ */
+export async function getObfuscatedIdentifierFromSignature(
+  plaintextIdentifier: string,
+  identifierType: string,
+  base64BlindedSignature: string,
+  blsBlindingClient: BlsBlindingClient
+): Promise<IdentifierHashDetails> {
+  debug('Retrieving unblinded signature')
+  const base64UnblindedSig = await blsBlindingClient.unblindAndVerifyMessage(base64BlindedSignature)
+  const sigBuf = Buffer.from(base64UnblindedSig, 'base64')
+
+  debug('Converting sig to pepper')
+  const pepper = getPepperFromThresholdSignature(sigBuf)
+  const obfuscatedIdentifier = getIdentifierHash(plaintextIdentifier, identifierType, pepper)
+  return {
+    plaintextIdentifier,
+    obfuscatedIdentifier,
+    pepper,
+    unblindedSignature: base64UnblindedSig,
+  }
+}
+
+export const getPrefixedIdentifier = (
+  plaintextIdentifier: string,
+  identifierPrefix: string
+): string => identifierPrefix + '://' + plaintextIdentifier
+
+export const getIdentifierHash = (
+  plaintextIdentifier: string,
+  identifierPrefix: string,
+  pepper: string
+): string => {
+  const value =
+    getPrefixedIdentifier(plaintextIdentifier, identifierPrefix) + PEPPER_SEPARATOR + pepper
+  return sha3(value) as string
```

Comment:
It's possible for many identifiers to contain underscores (e.g. Twitter) which raises the possibility of domain collisions. Let's use a simply trick, which is to hash the identifier, then tack on the pepper to the hash and hash the whole thing.
```suggestion
  const value = (sha3(getPrefixedIdentifier(plaintextIdentifier, identifierPrefix)) as string) + PEPPER_SEPARATOR + pepper
  return sha3(value) as string
```

Reply:
This break backwards compatibility :(

Reply:
putting this modification behind a conditional statement checking if the identifier is a phone number, similar to how we're doing blinding.